data/mods/DinoMod/monster_attacks.json

```diff
@@ -1,4 +1,88 @@
+  },
+  {
+    "type": "monster_attack",
+    "attack_type": "melee",
+    "id": "teeth_rip",
+    "cooldown": 25,
+    "move_cost": 150,
+    "damage_max_instance": [ { "damage_type": "cut", "amount": 15, "armor_penetration": 1, "armor_multiplier": 0.5 } ],
+    "hitsize_min": 4,
+    "effects": [ { "id": "bleed", "duration": 200, "affect_hit_bp": true }, { "id": "venom_pain", "duration": 1000 } ],
+    "hit_dmg_u": "%1$s rips your %2$s with its sharp teeth!",
+    "hit_dmg_npc": "%1$s rips a gash into <npcname> with its sharp teeth!",
+    "miss_msg_u": "%1$s tries to rip your %2$s, but you dodge!",
+    "miss_msg_npc": "%1$s tries to rip a gash into <npcname>, but they dodge!",
+    "no_dmg_msg_u": "%1$s tries to rip your %2$s with its sharp teeth, but fails to penetrate your armor!",
+    "no_dmg_msg_npc": "%1$s tries to rip a gash into <npcname>, but fails to penetrate their armor!",
+    "required_effects_any": [ "grabbing" ],
+    "dodgeable": false,
+    "blockable": true
+  },
+  {
+    "type": "monster_attack",
+    "attack_type": "melee",
+    "id": "teeth_puncture",
+    "cooldown": 15,
+    "move_cost": 210,
+    "range": 2,
+    "damage_max_instance": [ { "damage_type": "stab", "amount": 50, "armor_penetration": 50, "armor_multiplier": 0.5 } ],
+    "hitsize_min": 4,
+    "effects_require_dmg": false,
+    "effects": [
+      { "id": "grabbed", "duration": 300, "intensity": [ 5, 15 ] },
+      { "id": "bleed", "duration": 300, "affect_hit_bp": true }
+    ],
+    "accuracy": 7,
+    "blockable": false,
+    "dodgeable": true,
+    "self_effect_always": [ { "id": "grabbing", "duration": 300 } ],
+    "hit_dmg_u": "%s punctures you with its long sharp teeth!",
+    "hit_dmg_npc": "%s punctures <npcname> in a powerful bite!",
+    "miss_msg_u": "%s tries to catch you in its teeth, but you dodge!",
+    "miss_msg_npc": "%s tries to close its teeth around <npcname>, but they dodge!",
+    "no_dmg_msg_u": "%s grabs you in its long sharp teeth!",
+    "no_dmg_msg_npc": "%s grabs <npcname> in its long sharp teeth!"
+  },
+  {
+    "type": "monster_attack",
+    "attack_type": "melee",
+    "id": "teeth_pull",
+    "cooldown": 25,
+    "move_cost": 150,
+    "range": 2,
+    "damage_max_instance": [ { "damage_type": "cut", "amount": 50, "armor_penetration": 50, "armor_multiplier": 0.5 } ],
+    "hitsize_min": 4,
+    "hit_dmg_u": "%1$s pulls your %2$s apart with its sharp teeth!",
+    "hit_dmg_npc": "%1$s pulls away a large chunk of flesh from <npcname> with its sharp teeth!",
+    "miss_msg_u": "%1$s tries to pull your %2$s apart, but you dodge!",
+    "miss_msg_npc": "%1$s tries to pull away a large chunk of flesh from <npcname>, but they dodge!",
+    "no_dmg_msg_u": "%1$s tries to pull your %2$s apart with its sharp teeth, but fails to penetrate your armor!",
+    "no_dmg_msg_npc": "%1$s tries to pull away a large chunk of flesh from <npcname>, but fails to penetrate their armor!",
+    "required_effects_any": [ "grabbing" ],
```

Comment:
You'll want to test this to make sure it triggers - IIRC grabbing is removed when the monster is no longer adjacent to an enemy. If you activate Debug Mod with the DF:MATTACK filter you'll get a readout of the whole logic in the sidebar.